homeassistant/components/modbus/modbus.py

```diff
@@ -356,7 +357,9 @@ def _pymodbus_call(self, unit, address, value, use_call):
         self._in_error = False
         return result
 
-    async def async_pymodbus_call(self, unit, address, value, use_call):
+    async def async_pymodbus_call(
+        self, unit, address, value, use_call
+    ) -> ModbusResponse:
         """Convert async to sync pymodbus call."""
         if self._config_delay:
             return None
```

Comment:
Method returns `None` here, but claims to return (non-optional) `ModbusResponse`.

Reply:
First this PR is about base_platform.py, not modbus.py. I have just added the needed changes in modbus.py to make base_platform.py work.

But I have changed the function (be aware that modbus.py have a lot of other strict errors, which will be addresses in a PR concerning modbus.py).